libbeat/cmd/instance/beat.go

```diff
@@ -979,11 +982,104 @@ func (b *Beat) makeOutputFactory(
 	}
 }
 
+func (b *Beat) reloadOutputOnCertChange(cfg config.Namespace) error {
+	// TODO (Tiago): find a proper name for this logger
+	logger := logp.L().Named("ssl.cert.reloader")
+	// Here the output is created and we have acces to the Beat struct (with the manager)
+	// as a workaround we can unpack the new settings and trigger the reload-watcher from here
+
+	// We get an output config, so we extract the 'SSL' bit from it
+	rawTLSCfg, err := cfg.Config().Child("ssl", -1)
+	if err != nil {
+		e, ok := err.(ucfg.Error)
+		if ok {
+			if errors.Is(e.Reason(), ucfg.ErrMissing) {
+				// if the output configuration does not contain a `ssl` section
+				// do nothing and return no error
+				return nil
+			}
+		}
+		return fmt.Errorf("could not extract the 'ssl' section of the output config: %w", err)
+	}
+
+	extendedTLSCfg := struct {
+		tlscommon.Config `config:",inline" yaml:",inline"`
+		Reload           cfgfile.Reload `config:"exit_on_cert_change" yaml:"exit_on_cert_change"`
+	}{
+		Reload: cfgfile.Reload{
+			Enabled: false,
+			Period:  time.Minute,
+		},
+	}
```

Comment:
I think it would be better to define it as a separate type.
Should we also define a `default` function for the `Reload` part in the `cfgfile` package?

It's kinda difficult to find these default values inside the function body here, if you need to, especially for a new comer.

Reply:
> I think it would be better to define it as a separate type.

Agreed, I'll move it somewhere else on this file

> Should we also define a default function for the Reload part in the cfgfile package?

Different parts of Beats use this `Reload` struct, I believe the default value for it depends on the feature rather on the struct itself.

Reply:
> Different parts of Beats use this Reload struct, I believe the default value for it depends on the feature rather on the struct itself.

I think it makes sense to return the whole `extendedTLSCfg` from a `defaultExtendedTLSCfg` defined in this file then. It should be semantically marked as `default` somehow, I believe this is the way :)